identity-iota/src/credential/validation_options.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Timestamp;
+use identity_did::verifiable::VerifierOptions;
+
+#[derive(Debug)]
+/// Options to declare validation criteria in
+/// [CredentialValidator::full_validation](super::CredentialValidator::full_validation()).
+#[non_exhaustive]
+pub struct CredentialValidationOptions {
+  /// Declares that the credential is **not** considered valid if it expires before this
+  /// [Timestamp].
+  pub earliest_expiry_date: Timestamp,
+  /// Declares that the credential is **not** considered valid if it was issued later than this
+  /// [Timestamp].
+  pub latest_issuance_date: Timestamp,
+  /// Declare that the credential's signature must be verified according to these
+  /// [VerifierOptions].
+  pub verifier_options: VerifierOptions,
+}
+
+impl Default for CredentialValidationOptions {
+  fn default() -> Self {
+    Self {
+      earliest_expiry_date: Timestamp::now_utc(),
+      latest_issuance_date: Timestamp::now_utc(),
```

Comment:
I think the timestamps should be optional and default to `Timestamp::now_utc` when the validation is actually performed.

Because `CredentialValidationOptions` is only immutably borrowed by `full_validation`, I can imagine this resulting in subtle bugs where the same options are re-used again and again without the timestamp ever changing.